### PR TITLE
Python static typing

### LINKED ISSUE #1316: Python static typing
### Brief Summary: 
Python is a dynamically typed language. The interpreter handled types implicitly. This article demonstrates how several annotation tools can be used to specify constraint data types of objects. Typing makes the code cleaner and maintainable.

### Key Takeaways: 
1. Static typing in python
2. Function Annotations
3. Variable Annotations
4. Basics of mypy type checker

### References: 
N/A

### PR DESCRIPTION
**Pull request by jerimkaura**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1316

### CODE REVIEW

**Comment by lalith1403** [comment]
I think you have uploaded the earlier article on hash tables. Can you check and upload the correct article.

**Comment by jerimkaura** [comment]
I have uploaded the right files

File: articles/python-static-typing/index.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+# print output
+print(square_list_members(square, range(10,20)))
+
+```
+
+```
+[100, 121, 144, 169, 196, 225, 256, 289, 324, 361]
+```
+
+**4.2. List**
```
Comment (lalith1403) [inline articles/python-static-typing/index.md:68]: Use H4 and beyond for further subheadings. 

```suggestion
#### 4.2 List
```

File: articles/python-static-typing/index.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+4. Selected annotations in Python `Typing` module
+   - Callable
+   - List
+   - Dict & Any
+   - Union
+5. Using mypy static type checker
+
+### 1. Introduction to type annotations
+
+Type annotations is a new feature added by [PEP 484](https://www.python.org/dev/peps/pep-0484/) to python version 3.5 and above. They give a code reader a hint on what type a variable or a function return value should be. Type annotations can be performed on variables and functions as well.
```
Comment (lalith1403) [inline articles/python-static-typing/index.md:19]:
```suggestion
Type annotations is a new feature added by [PEP 484](https://www.python.org/dev/peps/pep-0484/) to python version 3.5 and above. They give the code reader hints on the return type of variable or function. Type annotations can be performed on variables and functions as well.
```

File: articles/python-static-typing/index.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+### 2. Variable annotations
+
+We achieve variable annotation by adding a semicolon and the type after declaring or initializing a variable. i.e `: <type>`. Like in the example below, instead of saying `name = "John Doe"` we specify the return type by writing `name: str = "John Doe"`
+
+```python
+name: str = "John Doe"
+print(name)
+
+```
```
Comment (lalith1403) [inline articles/python-static-typing/index.md:29]: Mention the output of the code as well.

File: articles/python-static-typing/index.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+### 3. Function annotations
+
+This is achieved by adding a forward arrow and indicating the function parameters' expected return type after the function parameters' closing bracket. i.e `-> <return type>`.
+
+```python
+
+def square(x: int) -> int:
+    return x * x
+
+```
```
Comment (lalith1403) [inline articles/python-static-typing/index.md:40]: Mention the output of the code as well. Do it for all code snippets in the article.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Make the changes requested, and we can move to the final round of review. This is a very informative article.  Well done :)

**Comment by jerimkaura** [comment]
I have done the changes requested.

**Comment by jerimkaura** [comment]
Fixed merge conflict on PR #1562

**Comment by hectorkambow** [comment]
Merge conflict

**Pull request closed without merging**